### ISSUE TITLE
Public key authentication in SFTP connector

### ISSUE THREAD

**Issue by zregvart** [syndesisio/syndesis#2660]
## This is a...

<pre><code>
[x] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem

For SFTP we support username and password authentication, we
should also support public key authentication.

## Expected behavior

Specify SSH private key instead of username and password to establish a connection to SFTP server.

## Tasks involved / Steps to Reproduce

1.
2.
3.
4.

**Comment by paoloantinori** [comment]
First step is to verify the functionality exists at Camel level

**Comment by oscerd** [comment]
@zregvart do you mean by using the keyPair parameter in Camel-sftp component?

**Comment by oscerd** [comment]
Or using privateKey as bytes as camel-sftp provides?

**Comment by zregvart** [comment]
@oscerd I'm thinking of `privateKey` parameter for `camel-sftp`, we probably need a connector property that will accept the openssh format key and convert it to `byte[]` for the `ftp2` component. I'd expect the users to provide the key in a text box by pasting something like:
```
-----BEGIN OPENSSH PRIVATE KEY-----
...
-----END OPENSSH PRIVATE KEY-----
```

**Comment by oscerd** [comment]
Understood, we'll need to set the preferredAuthentications in that case with public_key has first choice.

**Comment by oscerd** [comment]
@paoloantinori @heiko-braun I don't think this will go in this Sprint, I'll take care of this for the next one. Totally forget about  it :-|

**Comment by oscerd** [comment]
@paoloantinori @heiko-braun I'll work on this in the next Sprint, Totally forget about it :-|

**Comment by oscerd** [comment]
Didn't have the time for this enhancement. Too much stuff also in sustaining team.

**Comment by oscerd** [comment]
I completely forget about this again. Sorry.
